mne/decoding/time_gen.py

```diff
@@ -407,90 +464,123 @@ def _predict_time_loop(X, estimators, cv, slices, predict_mode,
-            start += n_test_epochs
-        X = X[all_test]  # XXX JRK: Still 12 % of cpu time.
-
-        # Check that training cv and predicting cv match
-        if (len(estimators) != len(cv)) or (cv.n != len(X)):
-            raise ValueError(
-                'When `predict_mode = "cross-validation"`, the training '
-                'and predicting cv schemes must be identical.')
-    elif predict_mode != 'mean-prediction':
-        raise ValueError('`predict_mode` must be a str, "mean-prediction" or'
-                         '"cross-validation"')
     n_epochs, _, n_times = X.shape
-
-    # Checks whether the GAT is based on contiguous window of lengths = 1
-    # time-sample, ranging across the entire times. In this case, we will
-    # be able to vectorize the testing times samples.
-    expected_start = np.arange(n_times)
-    is_single_time_sample = np.array_equal([ii for sl in slices for ii in sl],
-                                           expected_start)
-    if is_single_time_sample:
-        # In simple mode, we avoid iterating over time slices.
-        slices = [slice(expected_start[0], expected_start[-1] + 1, 1)]
-    elif _warn_once.get('vectorization', True):
-        warn('not vectorizing predictions across testing times, '
-             'using a time window with length > 1')
-        _warn_once['vectorization'] = False
-    # Iterate over testing times. If is_single_time_sample, then 1 iteration.
-    y_pred = list()
-    for t, indices in enumerate(slices):
-        # Vectoring chan_times features in case of multiple time samples given
-        # to the estimators.
-        if not is_single_time_sample:
-            X_pred = X[:, :, indices].reshape(n_epochs, -1)
-        else:
+    n_train = len(estimators)
+    n_test = [len(test_t_idxs) for test_t_idxs in train_times]
+
+    # Loop across training times (i.e. estimators)
+    y_pred = None
+    for train_t_idx, (estimator_cv, test_t_idxs) in enumerate(
+            zip(estimators, train_times)):
+        # Checks whether predict is based on contiguous windows of lengths = 1
+        # time-sample, ranging across the entire times. In this case, we will
+        # be able to vectorize the testing times samples.
+        # Set expected start time if window length == 1
+        start = np.arange(n_times)
+        contiguous_start = np.array_equal([sl[0] for sl in test_t_idxs], start)
+        window_lengths = np.unique([len(sl) for sl in test_t_idxs])
+        vectorize_times = (window_lengths == 1) and contiguous_start
+        if vectorize_times:
+            # In vectorize mode, we avoid iterating over time test time indices
+            test_t_idxs = [slice(start[0], start[-1] + 1, 1)]
+        elif _warn_once.get('vectorization', True):
+            warn('not vectorizing predictions across testing times, using a '
+                 'time window with length > 1')
+            _warn_once['vectorization'] = False
```

Comment:
Consider using a more informative warning - the problem here is speed, right? So it could say: ```Due to a time window with length > 1, unable to vectorize across testing times. This leads to slower predictions compared to the length == 1 case.```

Right?